### PR TITLE
test: Create 4 test cases to POST Attendees

### PR DESCRIPTION
**Pull request by viniciusvieira00**
## What does this PR do?

This PR adds tests for the `attendees` service in the API. Tests were created to cover the main control flows and validations of the service, such as access permissions and data integrity. The tests were developed using the MC/DC (Modified Condition/Decision Coverage) technique to ensure comprehensive coverage of critical conditions and decisions.

## Mandatory Tasks (DO NOT REMOVE)

- [x] I have self-reviewed the code (A decent size PR without self-review might be rejected).
- [x] N/A I have added a Docs issue [here](https://github.com/calcom/docs/issues/new) if this PR makes changes that would require a [documentation change](https://docs.cal.com). If N/A, write N/A here and check the checkbox. **N/A**
- [x] I confirm automated tests are in place that prove my fix is effective or that my feature works.

## How should this be tested?

To test the changes:

1. **Environment Variables**: Ensure all necessary environment variables for running the tests are correctly configured.
2. **Minimal Test Data**: Use the minimal test data provided in the test files to validate the scenarios.
3. **Happy Path**:
   - Verify a `403 Forbidden` response when the user is not an admin and does not have a booking.
   - Verify a `200 OK` response when the user is an admin.
   - Verify a `200 OK` response when the user is not an admin but has a booking.
   - Verify a `400 Bad Request` response when the request body is invalid.
4. **Other Information**: Ensure all input and output scenarios are covered as described in the `api/v1/test/lib/attendees/_post.test.ts` file.
5. Run yarn test api and see the results like below:
Screenshot 2024-07-21 at 17 25 11

### CODE REVIEW

**Review by anikdhabal — COMMENTED** [review]
@viniciusvieira00 one test is failing and also test-insights file is also empty

File: apps/api/v1/test/lib/attendees/_post.test.ts [hunk 1]
```diff
@@ -0,0 +1,115 @@
...
+          email: "test@example.com",
+          name: "Test User",
+          timeZone: "UTC",
+        },
+      });
+
+      prismaMock.booking.findFirst.mockResolvedValue(null);
+
+      req.userId = 123;
+      req.isAdmin = false;
```
Comment (anikdhabal) [inline apps/api/v1/test/lib/attendees/_post.test.ts:29]: I think we can remove this and do not need to specify that it is false

**Review by anikdhabal — COMMENTED** [review]
Hey, shouldn't this be `req.isSystemWideAdmin` instead of `req.isAdmin`? Pls check it once

File: apps/api/v1/test/lib/attendees/_post.test.ts (hunk 1 above)
Comment (viniciusvieira00) [reply apps/api/v1/test/lib/attendees/_post.test.ts:29]: That is correct, adjusted and still working, isAdmin as undefined will work as expected too

**Comment by viniciusvieira00** [comment]
> Hey, shouldn't this be `req.isSystemWideAdmin` instead of `req.isAdmin`? Pls check it once

Did this correction, everything working fine now

**Comment by anikdhabal** [comment]
@viniciusvieira00 Why are you committing the test-insights file?

**Comment by viniciusvieira00** [comment]
> @viniciusvieira00 Why are you committing the test-insights file? Also, instead of commenting `req.isAdmin`, it would be cleaner if you removed it

Was just a mistake the test-insights file on the commit, removed now, and also removed the commented line req.isAdmin=false, making the code cleaner. Please take a look!

**Review by anikdhabal — APPROVED** [review]
LGTM! thanks for your work

**Pull request merged**